src/logic/safe/store/actions/fetchSafe.ts

```diff
@@ -23,10 +23,16 @@ export const buildSafe = async (safeAddress: string): Promise<SafeRecordProps> =
   // setting `loadedViaUrl` to false, as `buildSafe` is called on safe Load or Open flows
   const safeInfo: Partial<SafeRecordProps> = { address, loadedViaUrl: false }
 
-  const [remote, localSafeInfo] = await allSettled<[SafeInfo | null, SafeRecordProps | undefined | null]>(
-    getSafeInfo(safeAddress),
-    getLocalSafe(safeAddress),
-  )
+  const [remote, localSafeInfo] = await Promise.all([
+    getSafeInfo(safeAddress).catch((err) => {
+      logError(Errors._605, err.message)
+      return null
+    }),
+    getLocalSafe(safeAddress).catch((err) => {
```

Comment:
can it throw?

I also noticed that at line 43 in this file where we spread `localSafeInfo` there's no check for undefined or null

Reply:
Removed the catch, it's already caught upstream. Thanks Mikhail!